Plugins/TextureOps/TextureOps.cs

```diff
@@ -394,6 +394,41 @@ public static Texture2D[] Slice( Texture sourceTex, int sliceTexWidth, int slice
 
 		return result;
 	}
+
+	public static Texture2D Crop(Texture sourceTex, int x, int y, int width, int height, TextureFormat format = TextureFormat.RGBA32, Options options = new Options())
+	{
+		if( sourceTex == null )
+			throw new ArgumentException( "Parameter 'sourceTex' is null!" );
+
+		Texture2D result = null;
+
```

Comment:
Adding these failsafe checks here would be great!

```suggestion

		if( width <= 0 || width > sourceTex.width )
			width = sourceTex.width;
		if( height <= 0 || height > sourceTex.height )
			height = sourceTex.height;

		if( leftOffset <= 0 )
			leftOffset = 0;
		else if( leftOffset + width > sourceTex.width )
			leftOffset = sourceTex.width - width;

		if( topOffset <= 0 )
			topOffset = 0;
		else if( topOffset + height > sourceTex.height )
			topOffset = sourceTex.height - height;
	
```